controllers/drplacementcontrol_controller.go

```diff
@@ -953,56 +1024,153 @@ func (d *DRPCInstance) switchToPreferredCluster(drState rmn.DRState) (bool, erro
 	d.setDRState(drState)
 	setMetricsTimerFromDRState(drState, timerStart)
 
-	// During failback or relocation, both clusters should be up and both must be secondaries before we proceed
-	ensured, err := d.processVRGForSecondaries()
-	if err != nil || !ensured {
-		return !done, err
+	// During failback/relocate, the preferredCluster is does not contain a VRG or the VRG is already
+	// secondary. We need to skip checking if the VRG for it is secondary to avoid messing up with the
+	// order of execution (it could be refactored better to avoid this complexity). IOW, if we first update
+	// VRG in all clusters to secondaries, then we call runPlacementTask. If runPlacementTask does not complete
+	// in one shot, then coming back to this loop will reset the preferredCluster to secondary again.
+	clusterToSkip := preferredCluster
+	if !d.ensureVRGIsSecondary(clusterToSkip) {
```

Comment:
(note) The one (more) case we need to handle is to ensure peer cluster is ready to accept the relocation, IOW cleaned up and a healthy secondary before we start. This is not handled even prior, hence just noting it here. This will incur issues as in the comment above, as we progress beyond the initial cleanup.

Reply:
Doesn't the VRG status reporting that the peer is secondary enough? I thought it is.